gammapy/datasets/metadata.py

```diff
@@ -0,0 +1,157 @@
+import logging
+from typing import Optional, Union
+import numpy as np
+from astropy.coordinates import SkyCoord
+from pydantic import ValidationError, validator
+from gammapy.utils.metadata import CreatorMetaData, MetaData
+
+__all__ = ["MapDatasetMetaData"]
+
+
+class MapDatasetMetaData(MetaData):
+    """Metadata containing information about the GTI.
+
+    Parameters
+    ----------
+    creation : `~gammapy.utils.CreatorMetaData`
+        the creation metadata
+    instrument : str
+        the instrument used during observation
+    telescope : str
+        The specific telescope subarray
+    observation_mode : str
+        observing mode
+    pointing : ~astropy.coordinates.SkyCoord
+        Telescope pointing direction
+    obs_ids : int
+        Observation ids stacked in the dataset
+    event_types : int
+        Event types used in analysis
+    optional : dict
+        Any other meta information
+    """
+
+    creation: Optional[CreatorMetaData]
+    instrument: Optional[str]
+    telescope: Optional[Union[str, list[str]]]
+    observation_mode: Optional[Union[str, list]]
+    pointing: Optional[Union[SkyCoord, list[SkyCoord]]]
+    obs_ids: Optional[Union[int, list[int]]]
```

Comment:
Are we sure `obs_id` can only be `int`?

Reply:
I had the same question about event types as well.